app/src/test/java/uk/ac/ebi/atlas/search/geneids/GeneIdSearchServiceTest.java

```diff
@@ -177,4 +186,143 @@ void ifQueryHasEmptySpeciesSearchAllSpecies() {
         verify(geneIdSearchDaoMock).searchGeneIds(searchString, "flybase_gene_id");
         verify(geneIdSearchDaoMock).searchGeneIds(searchString, "wbpsgene");
     }
+
+    @Test
+    void whenEmptyQueryFieldsGivenThenReturnsError() {
+        var requestParams = new LinkedMultiValueMap<String, String>();
+        final String emptyValue = "";
+        var generalCategory = "q";
```

Comment:
`q` is also part of the `VALID_QUERY_FIELDS` list, so that would not fit these tests.
I would leave these tests as they are now.

Reply:
Yes. My observation was more that any single field (or more than one) with only empty values, will (correctly) throw an exception. There’s nothing particular about `q` or `symbol` in this test and the next one in that regard. So I was thinking on picking a random field here and picking two random fields in the next one.

Reply:
Ok, because you wrote `remaining tests` previously. You only would like me to use it in those tests that throwing errors because of only empty values.

Reply:
Anyway, as you accepted the PR, I merged it and do something with this in my next PR.